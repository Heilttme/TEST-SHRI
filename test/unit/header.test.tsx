import { getAllByTestId, render } from "@testing-library/react";
import { initNewState, item1 } from "./common";
import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { Application } from "../../src/client/Application";
import events from "@testing-library/user-event";

it("cart has a badge with items count", () => {
  const testStore = initNewState({
    cart: { [item1.id]: { name: item1.name, count: 1, price: item1.price } },
  });

  const { getByText } = render(
    <MemoryRouter initialIndex={0} initialEntries={["/catalog"]}>
      <Provider store={testStore}>
        <Application />
      </Provider>
    </MemoryRouter>
  );

  const cartWithBadge = getByText(`Cart (1)`)

  expect(cartWithBadge).not.toBe(undefined);
});

it("cart has no badge with empty cart", () => {
  const testStore = initNewState({
    cart: { },
  });

  const { getByText } = render(
    <MemoryRouter initialIndex={0} initialEntries={["/catalog"]}>
      <Provider store={testStore}>
        <Application />
      </Provider>
    </MemoryRouter>
  );

  const cartWithBadge = getByText(`Cart`)

  expect(cartWithBadge).not.toBe(undefined);
});

function resizeWindow(width: number, height: number) {
  window.innerWidth = width;
  window.innerHeight = height;
  window.dispatchEvent(new Event('resize'));
}


it("mobile navbar collapses on any nav list click", async () => {
  resizeWindow(800, 400)

  const testStore = initNewState({
    cart: { },
  });

  const { container, asFragment } = render(
    <MemoryRouter initialIndex={0} initialEntries={["/catalog"]}>
      <Provider store={testStore}>
        <Application />
      </Provider>
    </MemoryRouter>
  );

  const hamburger = container.getElementsByClassName("navbar-toggler-icon")[0]
  const catalogLink = container.getElementsByClassName("nav-link")[0]

  await events.click(hamburger)
  await events.click(catalogLink)

  expect(asFragment()).toMatchSnapshot();
});

