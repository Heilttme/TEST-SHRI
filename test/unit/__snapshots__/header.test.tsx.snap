// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mobile navbar collapses on any nav list click 1`] = `
<DocumentFragment>
  <div
    class="Application"
  >
    <nav
      class="navbar navbar-expand-sm navbar-light bg-light"
    >
      <div
        class="container"
      >
        <a
          class="Application-Brand navbar-brand"
          href="/"
        >
          Kogtetochka store
        </a>
        <button
          aria-label="Toggle navigation"
          class="Application-Toggler navbar-toggler"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="Application-Menu collapse navbar-collapse"
        >
          <div
            class="navbar-nav"
          >
            <a
              aria-current="page"
              class="active nav-link"
              href="/catalog"
            >
              Catalog
            </a>
            <a
              class="nav-link"
              href="/delivery"
            >
              Delivery
            </a>
            <a
              class="nav-link"
              href="/contacts"
            >
              Contacts
            </a>
            <a
              class="nav-link"
              href="/cart"
            >
              Cart
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="container pt-4"
    >
      <div
        class="Catalog"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h1>
              Catalog
            </h1>
          </div>
        </div>
        <div
          class="row"
        >
          LOADING
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
